tests/components/bond/common.py

```diff
@@ -12,19 +13,32 @@
 from tests.async_mock import patch
 from tests.common import MockConfigEntry, async_fire_time_changed
 
-MOCK_HUB_VERSION: dict = {"bondid": "test-bond-id"}
+
+def patch_setup_entry(domain: str):
+    """Patch async_setup_entry for specified domain."""
+    return patch(f"homeassistant.components.bond.{domain}.async_setup_entry")
 
 
 async def setup_bond_entity(
-    hass: core.HomeAssistant, config_entry: MockConfigEntry, hub_version=None
+    hass: core.HomeAssistant,
+    config_entry: MockConfigEntry,
+    *,
+    patch_version=False,
+    patch_device_ids=False,
+    patch_platforms=False,
 ):
     """Set up Bond entity."""
-    if hub_version is None:
-        hub_version = MOCK_HUB_VERSION
-
     config_entry.add_to_hass(hass)
 
-    with patch("homeassistant.components.bond.Bond.version", return_value=hub_version):
+    with patch_bond_version() if patch_version else nullcontext(), patch_bond_device_ids() if patch_device_ids else nullcontext(), patch_setup_entry(
```

Comment:
i wonder if itd be cleaner to pass the option to your patch helper and let it decide to return null context rather than all these ifs inline with

Reply:
maybe something like

```python
with patch_bond_version(enabled=patch_version)
```

the enabled kwarg can default to True but if False it could returns null context

Reply:
this is a good idea, done.